### PR TITLE
Do not log warnings about proxy-related env vars

### LINKED ISSUE #10693: Unknown environment variable: DD_PROXY_NO_PROXY, DD_PROXY_HTTP and DD_PROXY_HTTPS
```bash
# agent version
Agent 7.33.0 - Commit: e6cfcb9 - Serialization version: v5.0.4 - Go version: go1.16.7
```

**Output of the info page (if this is a bug)**

```console
2022-01-28 02:13:08 UTC | CORE | DEBUG | (pkg/util/log/log.go:600 in func1) | Loaded features: []
2022-01-28 02:13:08 UTC | CORE | DEBUG | (pkg/util/log/log.go:600 in func1) | maxprocs: Leaving GOMAXPROCS=1: CPU quota undefined
2022-01-28 02:13:08 UTC | CORE | INFO | (pkg/util/log/log.go:620 in func1) | runtime: final GOMAXPROCS value is: 1
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_NO_PROXY
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_HTTP
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_HTTPS
2022-01-28 02:13:08 UTC | CORE | INFO | (pkg/util/log/log.go:620 in func1) | Found 'DD_PROXY_HTTP' env var, using it for the Agent proxy settings
2022-01-28 02:13:08 UTC | CORE | INFO | (pkg/util/log/log.go:620 in func1) | Found 'DD_PROXY_HTTPS' env var, using it for the Agent proxy settings
2022-01-28 02:13:08 UTC | CORE | INFO | (pkg/util/log/log.go:620 in func1) | Found 'DD_PROXY_NO_PROXY' env var, using it for the Agent proxy settings
2022-01-28 02:13:08 UTC | CORE | INFO | (pkg/util/log/log.go:620 in func1) | Features detected from environment: kubernetes
2022-01-28 02:13:08 UTC | CORE | INFO | (cmd/agent/app/run.go:250 in StartAgent) | Starting Datadog Agent v7.33.0
2022-01-28 02:13:08 UTC | CORE | DEBUG | (cmd/agent/app/run.go:299 in StartAgent) | Health check listening on port 5555
```

**Describe what happened:**

When setting `DD_PROXY_HTTP`, `DD_PROXY_HTTPS` and `DD_PROXY_NO_PROXY` environment variables on datadog agent container, the following warning messages are always written at startup time after upgrading datadog agent from v7.32.3 to v7.33.0. It seems to be related to #9646 and #10514.

```console
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_NO_PROXY
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_HTTP
2022-01-28 02:13:08 UTC | CORE | WARN | (pkg/util/log/log.go:640 in func1) | Unknown environment variable: DD_PROXY_HTTPS
```

**Describe what you expected:**

No warning messages should output when setting known environment variables.

**Steps to reproduce the issue:**

Apply the following manifest to Kubernetes cluster. The datadog-agent cannot startup successfully but you can see the logs including unknown environment variable warning messages.

```yaml
apiVersion: v1
data:
  datadog.yaml: |
    ## Provides autodetected defaults, for kubernetes environments,
    ## please see datadog.yaml.example for all supported options
kind: ConfigMap
metadata:
  name: datadog-yaml
  namespace: datadog
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: datadog
  namespace: datadog
spec:
  replicas: 1
  selector:
    matchLabels:
      app: datadog
      type: deployment
  template:
    metadata:
      labels:
        app: datadog
        type: deployment
      name: datadog
    spec:
      containers:
      - command: ["agent", "run"]
        env:
        - name: KUBERNETES
          value: "yes"
        - name: DD_LOG_LEVEL
          value: DEBUG
        - name: DD_DOGSTATSD_NON_LOCAL_TRAFFIC
          value: "true"
        - name: DD_HEALTH_PORT
          value: "5555"
        - name: DD_PROXY_HTTP
          value: "*****"
        - name: DD_PROXY_HTTPS
          value: "*****"
        - name: DD_PROXY_NO_PROXY
          value: 10.0.0.0/8
        - name: DD_APM_NON_LOCAL_TRAFFIC
          value: "true"
        - name: DD_ENABLE_PAYLOADS_EVENTS
          value: "false"
        - name: DD_ENABLE_PAYLOADS_SERIES
          value: "false"
        - name: DD_ENABLE_PAYLOADS_SERVICE_CHECKS
          value: "false"
        - name: DD_ENABLE_PAYLOADS_SKETCHES
          value: "false"
        image: datadog/agent:7.33.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 6
          httpGet:
            path: /health
            port: 5555
            scheme: HTTP
          initialDelaySeconds: 15
          periodSeconds: 15
          successThreshold: 1
          timeoutSeconds: 5
        name: agent
        volumeMounts:
        - mountPath: /etc/datadog-agent
          name: datadog-yaml
      volumes:
      - configMap:
          defaultMode: 420
          name: datadog-yaml
        name: datadog-yaml
```

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Silences warnings about DD_PROXY_NO_PROXY, DD_PROXY_HTTP and DD_PROXY_HTTPS env vars

### Motivation

#10693 

### Additional Notes

I'd like to backport this to 7.34.

### Possible Drawbacks / Trade-offs

None

### Describe how to test/QA your changes

Set DD_PROXY_HTTP and start the agent; you should see no log messages.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -135,6 +135,29 @@ unknown_key.unknown_subkey: true
...
+			var exp []string
+			if unknown {
+				exp = append(exp, v)
+			}
+			assert.Equal(t, exp, findUnknownEnvVars(Mock()))
+		}
+	}
+	t.Run("DD_API_KEY", test("DD_API_KEY", false))
+	t.Run("DD_SITE", test("DD_SITE", false))
+	t.Run("DD_UNKOWN", test("DD_UNKOWN", true))
```
Comment (sgnn7) [inline pkg/config/config_test.go:154]: I think the linter is right on this one
```suggestion
	t.Run("DD_UNKNOWN", test("DD_UNKNOWN", true))
```

File: releasenotes/notes/silence-env-var-warning-363dca8f70c17f92.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The Agent no longer logs purious warnings regarding environment variables
```
Comment (sgnn7) [inline releasenotes/notes/silence-env-var-warning-363dca8f70c17f92.yaml:11]:
```suggestion
    The Agent no longer logs spurious warnings regarding proxy-related environment variables
```

**Review by sgnn7 — APPROVED** [review]
@djmitche 2 tiny nits but LGTM and kudos on the turnaround time!

**Review by ogaca-dd — APPROVED** [review]

**Comment by djmitche** [comment]
OK, I think everything that could have failed in CI has passed here (windows-unit-tests being the longest).

**Pull request merged**